src/index.js

```diff
@@ -9,19 +9,21 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
-const request = require('request-promise-native');
+const { fetch } = require('@adobe/helix-fetch');
 const { wrap: status } = require('@adobe/helix-status');
 const { wrap } = require('@adobe/openwhisk-action-utils');
 const { logger } = require('@adobe/openwhisk-action-logger');
 const { epsagon } = require('@adobe/helix-epsagon');
+const querystring = require('querystring');
 const range = require('range_check');
 const { embed } = require('./embed.js');
 
-const ips = request('https://api.fastly.com/public-ip-list', { json: true });
+const ipList = fetch('https://api.fastly.com/public-ip-list')
```

Comment:
what happens if this throws? then you entire action can't be loaded and ends up in a weird state, I guess.

Reply:
the original code does the same thing. I guess the assumption is that the request will always go through? I think it'll fail at the point where you try to do object destructuring; and that should be enough for developers? Users?

Reply:
I just tested the action; faked a 401 response, and it doesn't really do anything. I think this is wrong? Isn't the whitelist a necessary safeguard?

Reply:
Same thing happens in the ordinary case